fragile test

### Confirmation

- [X] I have checked for similar issues.

### Describe the bug

There seems to be a fragile test error during cleanup, this is the test failing https://github.com/hemilabs/heminetwork/actions/runs/9016650152/job/24773648158?pr=109

I believe this is the error, I think that the database is being closed before its consumers
```
2024-05-09 11:36:19 ERROR bfg bfg.go:215 Failed to handle handle l2 keystones request request [::1]:40150: error getting l2 keystones: sql: database is closed [1715254579:651ddc8d25b44529]
2024-05-09 11:36:19 ERROR bfg bfg.go:225 Failed to handle handle l2 keystones request request: protocol write failed: failed to write JSON message: failed to marshal JSON: failed to write msg: failed to acquire lock: context canceled
--- FAIL: TestL2Keystone (1.22s)
    e2e_ext_test.go:785: Creating test database e2e_ext_test_db__4536514
    e2e_ext_test.go:785: Applying SQL file 0001.sql
    e2e_ext_test.go:785: Applying SQL file 0002.sql
    e2e_ext_test.go:785: Applying SQL file 0003.sql
    e2e_ext_test.go:785: Applying SQL file 0004.sql
    e2e_ext_test.go:785: Applying SQL file 0005.sql
    e2e_ext_test.go:785: Applying SQL file 0006.sql
    e2e_ext_test.go:785: Database version: 6
    e2e_ext_test.go:115: connecting to http://localhost:43809/v1/ws/private
    e2e_ext_test.go:115: connecting to http://localhost:39901/v1/ws/public
    e2e_ext_test.go:864: unexpected command bfgapi-l2-keystones-notification
    e2e_ext_test.go:222: Removing test database e2e_ext_test_db__4536514
```

### Expected behaviour

test should not be fragile

### Environment

Version: <!-- Hemi Network v1.2.3 -->
Operating System: <!-- e.g. Windows 11, macOS 14.2.1, Ubuntu 24.04 -->
Architecture: <!-- amd64, arm64 -->


### Additional Information

_No response_

--- a/e2e/e2e_ext_test.go
+++ b/e2e/e2e_ext_test.go
@@ -855,13 +855,26 @@ func TestL2Keystone(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	command, _, response, err := bfgapi.Read(ctx, bws.conn)
-	if err != nil {
-		t.Fatal(err)
-	}
+	var response any
+	var command protocol.Command
 
-	if command != bfgapi.CmdL2KeystonesResponse {
-		t.Fatalf("unexpected command %s", command)
+	for {
+		command, _, response, err = bfgapi.Read(ctx, bws.conn)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// there is a chance we get notifications from the L2KeystonesInsert
+		// call above, if they haven't been broadcast yet.  ignore those.
+		if command == bfgapi.CmdL2KeystonesNotification {
+			continue
+		}
+
+		if command == bfgapi.CmdL2KeystonesResponse {
+			break
+		} else {
+			t.Fatalf("unexpected command %s", command)
+		}
 	}
 
 	l2KeystonesResponse := response.(*bfgapi.L2KeystonesResponse)

--- a/e2e/e2e_ext_test.go
+++ b/e2e/e2e_ext_test.go
@@ -870,11 +870,11 @@ func TestL2Keystone(t *testing.T) {
 			continue
 		}
 
-		if command == bfgapi.CmdL2KeystonesResponse {
-			break
-		} else {
+		if command != bfgapi.CmdL2KeystonesResponse {
 			t.Fatalf("unexpected command %s", command)
 		}
+
+		break
 	}
 
 	l2KeystonesResponse := response.(*bfgapi.L2KeystonesResponse)

